### PR TITLE
feat: template panel

### PR DESCRIPTION
**Pull request by storyworks**
# Description

This is the onboarding panel with the new create a custom journey button (dependant on Tanner's PRs) and templates.

[- Link to Basecamp Todo
](https://3.basecamp.com/3105655/buckets/31822575/todos/5909445092)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Clicking on add a custom journey creates a new journey in the journey list
- [ ] Clicking on See all or See all templates button takes you to template library
- [ ] Clicking on each template takes you to the templates details page.
- [ ] The templates in the list have ids that match 
image
- [ ] The mobile side panel opens in a drawer when clicking the "Add" fab.
- [ ] The close drawer button next to the Panel name closes the drawer.

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged into main

### CODE REVIEW

**Review by tanflem — COMMENTED** [review]
Everything looked good in testing except when clicking the create journey button there is a long pause before it transitions the user into the journey editor where they can still interact with the screen. Seems like a UX decision on how to fix it. Whether to disable all buttons or have a load screen after the button is clicked. 

https://user-images.githubusercontent.com/17752008/228920198-d7eea49c-7fe4-49be-99e9-e9cb4eaa6b11.mov

File: apps/journeys-admin/src/components/OnboardingPanelContent/OnboardingPanelContent.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        <ContainedIconButton
+          label="Create Custom Journey"
+          thumbnailIcon={<ViewCarouselIcon />}
+          onClick={handleCreateJourneyClick}
+          loading={loading}
+        />
+      </SidePanelContainer>
+      <SidePanelContainer border={false}>
+        <Stack direction="row" justifyContent="space-between">
+          <Typography variant="subtitle1">Use Template</Typography>
```
Comment (csiyang) [inline apps/journeys-admin/src/components/OnboardingPanelContent/OnboardingPanelContent.tsx:86]:
```suggestion
          <Typography variant="subtitle1">{t(Use Template)}</Typography>
```

File: apps/journeys-admin/src/components/OnboardingPanelContent/OnboardingPanelContent.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+  const handleTemplateClick = (journeyId?: string): void => {
+    if (journeyId != null) void router.push(`/templates/${journeyId}`)
+  }
+
+  return (
+    <>
+      <SidePanelContainer>
+        <ContainedIconButton
+          label="Create Custom Journey"
```
Comment (csiyang) [inline apps/journeys-admin/src/components/OnboardingPanelContent/OnboardingPanelContent.tsx:78]:
```suggestion
          label={t("Create Custom Journey")}
```

File: apps/journeys-admin/src/components/OnboardingPanelContent/OnboardingPanelContent.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+              title={template.template.title}
+              description={template.template.description ?? ''}
+              image={template.template.primaryImageBlock?.src ?? ''}
+              overline={t('Template')}
+              border
+              onClick={() => handleTemplateClick(template?.template?.id)}
+            />
+          )
+        )
+      })}
```
Comment (csiyang) [inline apps/journeys-admin/src/components/OnboardingPanelContent/OnboardingPanelContent.tsx:114]: shouldn't need the return statement right
```suggestion
      {templates.map((template, index) => (
          {template?.template != null && (
            <MediaListItem
              key={template.template.id}
              loading={loadingTemplates[index]}
              title={template.template.title}
              description={template.template.description ?? ''}
              image={template.template.primaryImageBlock?.src ?? ''}
              overline={t('Template')}
              border
              onClick={() => handleTemplateClick(template?.template?.id)}
            />
          )
        )
      })}
```

**Comment by storyworks** [comment]
> Everything looked good in testing except when clicking the create journey button there is a long pause before it transitions the user into the journey editor where they can still interact with the screen. Seems like a UX decision on how to fix it. Whether to disable all buttons or have a load screen after the button is clicked.

I noticed that too! There is a loading icon that spins in the thumbnail but yeah possibly not obvious enough. 
Would be a good idea to disable the "Create custom journey" button until loading is finished (it's a common feedback convention) so we don't accidentally create 2 (as shown in your video). But yes agreed, that still doesn't solve the long loading between click and redirect so will be good to ask them.

**Review by csiyang — CHANGES_REQUESTED** [review]
Nice one! Just some minor changes

File: apps/journeys-admin/src/components/ContainedIconButton/ContainedIconButton.tsx [hunk 4]
```diff
@@ -31,7 +31,7 @@ export function ContainedIconButton({
 }: ContainedIconProps): ReactElement {
   return (
     <Card variant="outlined" sx={{ borderRadius: 2 }}>
-      <CardActionArea onClick={onClick}>
+      <CardActionArea onClick={onClick} disabled={loading}>
```
Comment (csiyang) [inline apps/journeys-admin/src/components/ContainedIconButton/ContainedIconButton.tsx:34]: test?

**Review by csiyang — APPROVED** [review]

**Pull request merged**